### PR TITLE
Only display recipient's objectives; cleanup after multirecipient reports

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change

2 changes 
- when a recipient is deleted from a report, the backend recieves that information and cleans up any goals that are attached to that recipient
- on the RTR, objectives are explicitly checked for belonging to that recipient, not only to that recipients activity reports

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-898

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Comment by thewatermethod** [comment]
@AdamAdHocTeam I've made some additional changes

**Comment by AdamAdHocTeam** [comment]
> @AdamAdHocTeam I've made some additional changes

Going from topics to goals I get the Topics to Goals page I get a Sequelize error can you repro?

image

**Comment by thewatermethod** [comment]
@AdamAdHocTeam I've made some additional changes

**Review by AdamAdHocTeam — COMMENTED** [review]
Found another possible issue not sure if its related to this change.

Create a report add two recipients and a goal then save. Note that the 'ActivityReportGoals' table has two entries for this report. Remove both recipients and save. Then add back two recipients and save again. I don't see the entry being put back in the AR goals table.

**Comment by thewatermethod** [comment]
> Create a report add two recipients and a goal then save. Note that the 'ActivityReportGoals' table has two entries for this report. Remove both recipients and save. Then add back two recipients and save again. I don't see the entry being put back in the AR goals table.

I adjusted our existing "findWhatsChanged" method to always push the goals to the backend if the activityRecipients have changed. This should ensure that the ActivityReportGoals table stays updated.

**Review by AdamAdHocTeam — COMMENTED** [review]
Another potential bug.

Create an AR with two recipients add a goal and save. Then remove both recipients and save. Then add back two recipients and save. If you check the goals table you will see there is a extra goal left behind. ie you should have two but there are three.

**Comment by thewatermethod** [comment]
> Create an AR with two recipients add a goal and save. Then remove both recipients and save. Then add back two recipients and save. If you check the goals table you will see there is a extra goal left behind. ie you should have two but there are three.

I tried this with a custom goal and an existing goal and could not recreate

**Comment by AdamAdHocTeam** [comment]
> > Create an AR with two recipients add a goal and save. Then remove both recipients and save. Then add back two recipients and save. If you check the goals table you will see there is a extra goal left behind. ie you should have two but there are three.
> 
> I tried this with a custom goal and an existing goal and could not recreate

When you add back the recipients make sure one of them is different than before.

**Comment by thewatermethod** [comment]
@AdamAdHocTeam this should be good for another look when you have a chance

File: frontend/src/pages/ActivityReport/index.js [hunk 1]
```diff
@@ -140,14 +140,34 @@ export const findWhatsChanged = (object, base) => {
       return accumulator;
     }
 
+    // this block intends to fix an issue where multi recipients are removed from a report
+    // after goals have been saved we pass up the removed recipients so that their specific links
+    // to the activity report/goals will be severed on the backend
+    if (current === 'activityRecipients' && !isEqual(base[current], object[current])) {
+      // eslint-disable-next-line max-len
+      const grantIds = object.activityRecipients.map((activityRecipient) => activityRecipient.activityRecipientId);
```
Comment (GarrettEHill) [inline frontend/src/pages/ActivityReport/index.js:148]: is there a reason the activityRecipientId's are being put into a variable called grantIds?

Comment (thewatermethod) [reply frontend/src/pages/ActivityReport/index.js:148]: those numbers, for whatever reason, are the grant ids, not the ids in the sense of activityRecipient.id

File: src/services/goals.js [hunk 2]
```diff
@@ -457,8 +448,79 @@ async function removeObjectives(currentObjectiveIds) {
...
+      id: objectiveIds,
+    },
+  });
+
+  return Goal.destroy({
+    where: {
+      id: goalIds,
+      onApprovedAR: false,
+    },
+  });
```
Comment (GarrettEHill) [inline src/services/goals.js:519]: we can't directly delete. we need to check the objective and goal to make sure there is no other reports its used on.

File: frontend/src/pages/ActivityReport/index.js (hunk 1 above)
Comment (GarrettEHill) [reply frontend/src/pages/ActivityReport/index.js:148]: what value does it have when the activity recipients are other entities? as they don't have grants.

File: src/services/goals.js (hunk 2 above)
Comment (thewatermethod) [reply src/services/goals.js]: Think this should be good now

**Pull request merged**